tiledb/sm/enums/datatype.h

```diff
@@ -347,6 +353,23 @@ inline bool datatype_is_time(Datatype type) {
       type == Datatype::TIME_AS);
 }
 
+/** Throws error if the input Datatype's enum is not between 0 and 40. */
+inline void ensure_datatype_is_valid(Datatype type) {
+  std::underlying_type_t<Datatype> datatype_enum{::stdx::to_underlying(type)};
```

Comment:
Can use `auto` safely here. No urgent need to change it.

Nit: Extraneous `::` before `stdx`.

Reply:
Note: using `::stdx` because compiler cannot resolve `stdx`